lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -2653,14 +2653,13 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
             // To accomplish this, we can set the switchover date between julian/gregorian
             // to the ECMAScript beginning of time, which is -8.64e15 according to ecma262 #sec-time-values-and-time-range
             UCalendar *cal = const_cast<UCalendar *>(udat_getCalendar(*dtf));
-            ucal_setGregorianChange(cal, -8.64e15, &status);
-
-            // status can be U_UNSUPPORTED_ERROR if the calendar isn't gregorian, which
-            // there does not seem to be a way to check for ahead of time in the C API
-            AssertOrFailFastMsg(U_SUCCESS(status) || status == U_UNSUPPORTED_ERROR, ICU_ERRORMESSAGE(status));
-
-            // If we passed the previous check, we should reset the status to U_ZERO_ERROR (in case it was U_UNSUPPORTED_ERROR)
-            status = U_ZERO_ERROR;
+            if (strcmp(ucal_getType(cal, &status), "gregorian") == 0)
```

Comment:
Thanks Jack! 👍

Reply:
I can't see much documentation for this; are we expected to free/release the string returned by `ucal_getType`? Or is it static?

Reply:
Any idea what the error scenarios are too? If `status` ends up not being success, is the returned string guaranteed to be nullptr or otherwise safe? Should we be checking the status code prior to using the returned value?

Reply:
I can assert right after getType, though I believe anything that returns a `const char *` returns a static string that shouldn't be freed.

Reply:
Yeah, I would agree that the documentation is rather lacking for this ICU API.

The API will return a static string that does not need to be free'ed. The only way `status` would be set to a failure state is if the incoming parameter is already a failure. The API itself doesn't modify the `status` variable.

I've filed a ticket on upstream ICU to improve the docs for this function:
https://unicode-org.atlassian.net/browse/ICU-20082
(Thanks for the feedback! :) )